WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/CardReaderSettingsViewModelsOrderedList.swift

```diff
@@ -0,0 +1,59 @@
+import Foundation
+
+/// Aggregates an ordered list of viewmodels, conforming to the viewmodel provider protocol. Priority is given to
+/// the first viewmodel in the list to return true for shouldShow
+///
+final class CardReaderSettingsViewModelsOrderedList: CardReaderSettingsPrioritizedViewModelsProvider {
+
+    private var viewModelsAndViews = [CardReaderSettingsViewModelAndView]()
+
+    var priorityViewModelAndView: CardReaderSettingsViewModelAndView? {
+        didSet {
+            onPriorityChanged?(priorityViewModelAndView)
+        }
+    }
+
+    var onPriorityChanged: ((CardReaderSettingsViewModelAndView?) -> ())?
+
+    init() {
+        /// Instantiate and add each viewmodel related to card reader settings to the
+        /// array. Viewmodels will be evaluated for shouldShow starting at the top
+        /// of the array. The first viewmodel to return true for shouldShow is given
+        /// priority, so viewmodels related to no-known-readers should come before viewmodels
+        /// that expect a connected reader, etc.
+        ///
+        viewModelsAndViews.append(
+            CardReaderSettingsViewModelAndView(
+                viewModel: CardReaderSettingsUnknownViewModel(
+                    didChangeShouldShow: onDidChangeShouldShow
+                ),
+                viewIdentifier: "CardReaderSettingsUnknownViewController"
+            )
+        )
+        viewModelsAndViews.append(
+            CardReaderSettingsViewModelAndView(
+                viewModel: CardReaderSettingsConnectedViewModel(
+                    didChangeShouldShow: onDidChangeShouldShow
+                ),
+                viewIdentifier: "CardReaderSettingsConnectedViewController"
+            )
+        )
+
+        /// And then immediately get a priority view if possible
+        reevaluatePriorityViewModelAndView()
+    }
+
+    private func onDidChangeShouldShow(_ : CardReaderSettingsTriState) {
+        reevaluatePriorityViewModelAndView()
+    }
+
+    private func reevaluatePriorityViewModelAndView() {
+        let newPriorityViewModelAndView = viewModelsAndViews.first(
```

Comment:
I am not sure if this is overthinking it too much, but `first` should return an optional, so maybe this could be written as (pseudocode)
```
guard let newThingy = thingies.first(), newThingy!= oldThingy else { return }

oldThingy = newThingy
```

It feels more "familiar" from a Swift point of view.

Reply:
Fixed in https://github.com/woocommerce/woocommerce-ios/pull/4123/commits/ec97ab87df9ba979a66affb8fc681c35edfa6f19

Reply:
In the case where `first` doesn't find any matching model and returns `nil`, shouldn't this replace the existing `priorityViewModelAndView` with that `nil`? In that case, I think the first attempt was correct.

Reply:
Ooooh. Yeah. I'll fix this in the next PR. Thank you @koke !